lib/land/trackers/api_tracker.rb

```diff
@@ -183,6 +180,30 @@ def page_view_path
         request && request.params['page_view_path']
       end
 
+      def device_width
+        request && request.params.dig('device_resolution', 'width')
+      end
+
+      def device_height
+        request && request.params.dig('device_resolution', 'height')
+      end
+
+      def device_orientation
+        request && request.params.dig('device_resolution', 'orientation')
+      end
+
+      def dark_mode
+        request && request.params.dig('color_scheme_preference', 'is_dark_mode')
+      end
+
+      def light_mode
+        request && request.params.dig('color_scheme_preference', 'is_light_mode')
+      end
+
+      def no_preference
+        request && request.params.dig('color_scheme_preference', 'is_no_preference')
+      end
```

Comment:
This can be refactored for succinctness:

```suggestion
      def device_width       = params.dig('device_resolution', 'width')
      def device_height      = params.dig('device_resolution', 'height')
      def device_orientation = params.dig('device_resolution', 'orientation')
      def dark_mode          = params.dig('color_scheme_preference', 'is_dark_mode')
      def light_mode         = params.dig('color_scheme_preference', 'is_light_mode')
      def no_preference      = params.dig('color_scheme_preference', 'is_no_preference')
         
      def params = request && request.params
```

Reply:
Also, if these are used in only one place, there is no need to pull into a function, as the param access is intent revealing. 🤔